### ISSUE TITLE
Allow to change font size

### ISSUE THREAD

**Issue by rafan** [keepassxreboot/keepassxc#621]
No current way to adjust font size in preference setting.

## Expected Behavior
Preference should allow user to change the font size.

## Current Behavior
No font size setting in preference. 

## Possible Solution
Allow user to set UI font size in preference setting.

## Steps to Reproduce (for bugs)
n/a

## Context
Would like to use larger font size.

## Your Environment
* KeePassXC version/commit used: 2.1.4 cb283bb95ae559d044415f426f841c27bff383f0
* Qt version (e.g. Qt 5.3): Qt 5.8.0
* Compiler (e.g. Clang++3.6.0): n/a
* Operating System and version: Windows 10

**Comment by O-Gopal** [comment]
Yes, please. For me the small font is a deal-breaker. 
KeepPassXC 2.2.0. Ubuntu 16.04.

**Comment by TheZ3ro** [comment]
@O-Gopal can you post a screenshot?

**Comment by droidmonkey** [comment]
There is a qt configuration tool that changes the default font size for all qt apps.

**Comment by O-Gopal** [comment]
![keepassxc--screenshot](https://user-images.githubusercontent.com/2028254/27644403-baddc764-5c23-11e7-8b41-3422b74c32b1.png)

Screenshot attached.

I found the QT5 configuration tool. But it's not in the Ubuntu repos, and at this point installing from a tarball is beyond my skill set. (Also: Changing the default font size for _all _ apps is not what I want to do.

Thank you.

**Comment by phoerious** [comment]
I edited your post. Please don't post screenshots of your actual password database.

**Comment by O-Gopal** [comment]
Thank you, phoerious. That was helpful. I've attached a new screenshot to that text.

**Comment by 5paceToast** [comment]
In the meanwhile, you can simply set the DPI scaling for QT to a custom value.

On windows 10, you'd do this [edit environment variables (searchable)] -> Environment Variables -> for $user.

The value you want to set it `QT_SCALE_FACTOR`.

**Comment by user23498723452** [comment]
Hoping to see progress on this.  The single font size is annoying.  I'm not very interested in hacking QT5.  Is this on the backlog or is this consider infeasible?

**Comment by droidmonkey** [comment]
This will be worked as part of v2.5.0 as it is addressing multiple accessibility concerns

**Comment by droidmonkey** [comment]
We actually received funding from a non-profit org to address accessibility issues, so you can expect results in this case :grin:

**Comment by sebast889** [comment]
@droidmonkey that's nice to hear about the funding. Is it possible to get some sort of .md file or wiki entry about any funding the project receives? I think that would be good for transparency sake

**Comment by hiredgunhouse** [comment]
+1
I'd love to switch to KeePassXC, but the font size is a deal breaker for me too.
Looking forward to release of 2.5.
Also thank you for your hard work!

**Comment by DanMan** [comment]
It wouldn't need a font size setting on Gnome, if it adhered to its font settings. I can change the fonts in the Gnome tweak tool all I want, but that doesn't change anything in KeePassXC. So maybe this ought to be a Qt<>Gtk interoperability fix.

**Comment by jnardone** [comment]
2 years and still can't do this? :(

**Comment by phoerious** [comment]
IMHO, this is something you should configure in your system. KeePassXC uses whatever font size you set there. If KeePassXC is too small, so is probably the rest of your desktop.

**Comment by vsund** [comment]
@phoerious at least for me, the font size in KeepassXC is smaller than the one in KeePassX (and also smaller than the rest of the system). I also tried using the previously mentioned `QT_SCALE_FACTOR` but the font gets blurry, so I still didn't made the switch to KeePassXC yet.

**Comment by phoerious** [comment]
Make sure you have correct font settings for non-GTK apps.

**Comment by vsund** [comment]
> Make sure you have correct font settings for non-GTK apps.

Whoa, you're totally right. I needed to update the settings which [make Qt look like GTK](https://wiki.archlinux.org/index.php/Uniform_look_for_Qt_and_GTK_applications#QGtkStyle) for working correctly with Qt 5.

To be honest, I didn't look much into _my_ settings because KeePassX/Qt4 worked as expected. Thanks for the nudge in the right direction ;)

**Comment by jnardone** [comment]
On Windows, there's not real utility for configuring global Qt settings.

**Comment by droidmonkey** [comment]
@jnardone you can create and define your own application wide stylesheet which can increase the font size. See here https://doc.qt.io/qt-5/qapplication.html#styleSheet-prop

**Comment by OLLI-S** [comment]
When you have KeePassXC running in Windows 10 and set the DPI in Windows to 150%, then there are many issues like:
- text is not enlarged (tree view, list of passwords, some values in the password-details panel, password history)
- elements are too small for the larger text (menu at the top, tabs in password-details panel, text og group boxes)
- edit fields are overlapping captions (when you edit the password)

When you set the DPI to 150%, close KeePassXC and start it again then some texts that were large before closing are now small (like text in the menu, text on tabs).

KeePassXC must support zooming without any display issues and also without the need to restart the application / restart the system).

**Comment by droidmonkey** [comment]
Everything you describe is a Qt concern, they are addressing it

**Comment by OLLI-S** [comment]
> Everything you describe is a Qt concern, they are addressing it

OK, did not know that.
So hopefully Qt is fixing this soon.

**Comment by regisz** [comment]
`Qt Settings` is solved my "too big" font problem.

**Comment by hi-ko** [comment]
I fixed this in ubuntu 18 by installing the qt5 config tool and the common Windows font:

* `sudo apt install qt5ct`
* install expected fonts (e.g. SegoeUI as Windows default UI font) 
* set `export QT_QPA_PLATFORMTHEME=qt5ct`
you should put this in your .profile
* run `qt5ct` and configure fonts you like

**Comment by droidmonkey** [comment]
We follow UI settings for the Operating System now in 2.6.0.

**Comment by O-Gopal** [comment]
Beautiful. Many thanks to the devs!

**Comment by mediaDS** [comment]
In Windows 10 the UI font seems to be ok but the font for notes is way too small  - no matter what I do in the OS. It is almost unreadable. There should be an option to change font size in the settings - and not qt5.

**Comment by LinuxOnTheDesktop** [comment]
> QT_SCALE_FACTOR [in Windows]

Some notes on this, for other users.

1) I had to create the string for the value.
2) The value of 1.25 was accepted and used.
3)  I did not need to log off and log on again, though perhaps that is because I used a program called 'Rapid Environment Editor' to set the value.

**Comment by LinuxOnTheDesktop** [comment]
> We follow UI settings for the Operating System now in 2.6.0.

Hmm. I have 2.6.1 on Windows 10. I have a 'custom scaling' of 125:

![image](https://user-images.githubusercontent.com/47056543/91309207-38b30600-e7a8-11ea-80eb-3432e7775c31.png)

(The image's blurriness is an artefact of my having scaled it to post it.) _Yet_, I had to use the environmental variables hack to make the text larger - large enough for me to read easily, and larger (I am fairly sure) than it was pre-version 2.6.

**Comment by LorenzoTa** [comment]
I've read this thread but I did not understand why it is not possible have a menu ( options->interface->font) entry to choose the font and its size.

the ENV trick works.. 
`cd "c:\Program Files\KeePassXC"`

`set QT_SCALE_FACTOR=1.25`

`KeePassXC.exe`

..but an appropriate menu for this configuration will be much better.

Win10 KeePassXC.exe v 2.6.1

Anyway thanks for this wonderful program!

**Comment by CYBRPVNK** [comment]
> 
> 
> I've read this thread but I did not understand why it is not possible have a menu ( options->interface->font) entry to choose the font and its size.
> 
> the ENV trick works..
> `cd "c:\Program Files\KeePassXC"`
> 
> `set QT_SCALE_FACTOR=1.25`
> 
> `KeePassXC.exe`
> 
> ..but an appropriate menu for this configuration will be much better.
> 
> Win10 KeePassXC.exe v 2.6.1
> 
> Anyway thanks for this wonderful program!

The Qt environment variable changes the scaling for all programs which use Qt renderer. KeepassXC might work fine with this scaling but it could break other programs. It might not even work in a different setup.

**Comment by awesomer** [comment]
In Windows 10, if one wants KeepassXC to start with scaled UI with the Start Menu shortcut, one can edit the "Target" to say :

> C:\Windows\System32\cmd.exe /c set QT_SCALE_FACTOR=1.25 && "C:\Program Files\KeePassXC\KeePassXC.exe"

It will, unfortunately, also pop up a CMD window, but you can immediately close this and KeePassXC will continue to run.

**Comment by peter-fb** [comment]
On Windows, the font size of the UI as a whole is OK, but the font size for notes is way too small.

**Comment by inkista** [comment]
Quick fix for Windows 10 users: **Settings** > **Basic Settings**, scroll down to **User Interface**, and select the **Use monospaced font for notes** checkbox.  The font size will now be proportional to font used in the rest of the UI.

For devs: the proportional font is not scaled properly vs. the monospace one:

![proportional](https://user-images.githubusercontent.com/81258790/112201464-81f7f680-8bcd-11eb-913b-7da6605edb50.png)

![monospace](https://user-images.githubusercontent.com/81258790/112201257-4a894a00-8bcd-11eb-9f13-abd2b9cdde43.png)

**Comment by tredondo** [comment]
> We follow UI settings for the Operating System now in 2.6.0.

One reason I think this may not be a comprehensive fix, is: users may have left the "UI settings for the OS" alone, because most other apps they use, have font scaling. I don't recall ever changing the font scaling in KDE, but LibreOffice, browsers, IDEs, and anything else I use with fonts that I want to scale up, has a setting to scale them up.

KeepassXC notes still look tiny. The font of the Notes field is visibly smaller than the font of the menu.

`QT_SCALE_FACTOR=1.25 keepassxc` works though, so I guess this issue won't be reopened.

**Comment by droidmonkey** [comment]
There was a bug in the notes when using monospaved fonts. That is fixed for next release.

**Comment by tredondo** [comment]
I've been using proportional fonts. This is what they look like with 1.25 QT scaling - maybe normal?

![image](https://user-images.githubusercontent.com/10043369/114726020-b3e60f80-9cd8-11eb-8f9a-df57ea06e912.png)

**Comment by droidmonkey** [comment]
Those look consistent and normal. Your scaling really depends on your base font size.

**Comment by Teknomancer** [comment]
Unfortunately using `QT_SCALE_FACTOR` (or other environment variables) on Windows is not optimal.

1. Setting it per-user, means it affects ALL QT apps (not just KeePassXC) and that is a problem in and of itself. It's not really an option if it breaks other Qt apps.

2. Setting it only for the KeePassXC shortcut by launching a `cmd.exe` with the environment variable and using the Run as "Minimize" window option is still not correct because while it launches fine, every time you minimize the KeePassXC window the `cmd.exe` window pops back up. Thus, even restricting the environment variable to KeePassXC leads to this sub-optimal behavior.

Please consider offering an option to choose the font and font size for UI elements.

**Comment by Bibelo** [comment]
Really surprised that this problem which affects so many users has not been solved after many years (five!). Same for regular Keepass

I'm using Keepass for note taking, and the Notes field is starting to be an eye sore

I agree that other programs also use the system font, but they allow you to dynamically change font size, from Chrome to WSL to Notepad++

**Comment by droidmonkey** [comment]
Is it a problem though? If you need bigger font sizes don't you need them everywhere? Why not set your whole operating system to be the right size? Are those other apps not respecting the operating system settings which is why they have specific font size choices?

Lastly, in 5 years not a single person has offered code to offer this option. To me that signals the operating system solution is perfectly acceptable to the vast majority.

**Comment by Bibelo** [comment]
Thanks for your answer. I image modifying code takes time.

I tried change the scale of the screen from 100% to 125% but the whole Windows everything is way too big. But I guess it won't be a problem soon when I enter old age and everything needs to be huge :-)

**Comment by CosmicToast** [comment]
For some reason, Github hasn't been letting me unsubscribe from this issue for 4 years now (I try whenever I get a couple of emails about it, despite having unsubscribed; oddly enough it works for other threads).
Either way, I just want to mention:
> Lastly, in 5 years not a single person has offered code to offer this option. To me that signals the operating system solution is perfectly acceptable to the vast majority.

This is a non sequitur: it does signal that whatever option people end up going with is easier to swallow than going out and working on a code solution, but it does not imply that the option people go with is the operating system solution.
This includes me and several other people I know (which is anecdotal, but sufficient as a counterexample to demonstrate the above generalization).

While userbase isn't a strictly important metric for an open source project (as it often fails to predict donations accurately), I did want to mention this regardless, because I do think keepassxc is an important project to have around, and would rather not have the development team (which droidmonkey is notable a part of) driving it blindly.

**Comment by JohnLGalt** [comment]
> 

Instead of enabling display scaling, did you try just text size adjustments?

For Windows 10, see this tutorial: https://www.tenforums.com/tutorials/112263-change-text-size-windows-10-a.html

For Windows 11, System settings --> Display --> click on the bar that has Scale in it (don't change the button value, click on the whole bar itself) --> Text size. (at this point the System settigns app should show `System > Display > Custom scaling` at the top)

HTH

**Comment by maprihoda** [comment]
This is a huge issue on Macs where there is no option for scaling the system fonts. 

For those of us who need to have as much real estate on an external monitor as possible (using higher resolution, which however makes the font sizes very small), fortunately, in many applications (even Finder!) we have the option to increase the fonts sizes: I do this in Finder, Pycharm, VSCode, Sublime Text, iTerm2, Email, Chrome/Firefox, etc. Not ideal, but at least something.

Big pity that this is not possible in keepassxc. Would be really great if you could revisit this issue.

**Comment by maprihoda** [comment]
I found a solution:
```
alias kp="QT_SCALE_FACTOR=1.4 /Applications/keepassxc.app/contents/MacOS/KeePassXC /Users/${whoami}/path/to/db.kdbx 2>/dev/null &"
```

**Comment by tredondo** [comment]
> Lastly, in 5 years not a single person has offered code to offer this option. To me that signals the operating system solution is perfectly acceptable to the vast majority.

No, it does not signal that _at all_. I'm stupefied that this illogical argument came from a dev, because it's invalid on so many counts.

First, the vast majority of KeePassXC users are not programmers, so they couldn't offered code even if they wanted to.

Second, even among the programmers, the vast majority don't know the OS internals and codebase to offer a code solution.

Third, it is assumes that a code solution exists.
